### DISCUSSION TITLE
ocrmypdf

### DISCUSSION THREAD

**Discussion by ferdiga** [nextcloud/all-in-one#5013] [Ideas] [answered]
Hi
IMO ocrmypdf should be included either in the container or in the OCR workflow package, as together with fulltext search it's an essential module (at least for my organisation)

To get it working I installed inside the container 
```
docker exec -it nextcloud-aio-nextcloud bash
apk add ocrmypdf
apk add tesseract-ocr
apk add tesseract-ocr-data-fra
apk add tesseract-ocr-data-deu
apk add tesseract-ocr-data-eng
```
but this is not a persistent solution and will vanish once the container gets updated 

would be happy to be advised to get a better solution.

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-add-os-packages-permanently-to-the-nextcloud-container

**Reply by ferdiga** [reply]
thanks for pointing out, but it does not work (or I made a mistake)

I got an error "ocrmypdf:latest not found"

it looks like that 
--env NEXTCLOUD_ADDITIONAL_APKS="ocrmypdf" looks for a docker container and not for a package.

**Comment by SasSam** [comment]
> Hi IMO ocrmypdf should be included either in the container or in the OCR workflow package, as together with fulltext search it's an essential module (at least for my organisation)
> 
> To get it working I installed inside the container
> 
> ```
> docker exec -it nextcloud-aio-nextcloud bash
> apk add ocrmypdf
> apk add tesseract-ocr
> apk add tesseract-ocr-data-fra
> apk add tesseract-ocr-data-deu
> apk add tesseract-ocr-data-eng
> ```
> 
> but this is not a persistent solution and will vanish once the container gets updated
> 
> would be happy to be advised to get a better solution.

Did you find a solution for this? I need the OCR as well. I'm new to nextcloud, so I don't know what I should do after I install these packages in the container. Could you write down you steps you did? How this will work in nextcloud?

**Reply by szaimen** [reply]
Hi, see https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-add-os-packages-permanently-to-the-nextcloud-container
